spec/devise_saml_authenticatable/saml_mapped_attributes_spec.rb

```diff
@@ -6,19 +6,28 @@
   let(:attribute_map_file) { File.join(File.dirname(__FILE__), '../support/attribute-map.yml') }
   let(:attribute_map) { YAML.load(File.read(attribute_map_file)) }
   let(:saml_attributes) do
-    {
+    OneLogin::RubySaml::Attributes.new({
       "first_name" => ["John"],
       "last_name"=>["Smith"],
       "email"=>["john.smith@example.com"]
-    }
+    })
   end
 
   describe "#value_by_resource_key" do
+    before do
+      @single_value_compatibility = OneLogin::RubySaml::Attributes.single_value_compatibility
+      OneLogin::RubySaml::Attributes.single_value_compatibility = false
+    end
+
+    after do
+      OneLogin::RubySaml::Attributes.single_value_compatibility = @single_value_compatibility
+    end
+
     RSpec.shared_examples "correctly maps the value of the resource key" do |saml_key, resource_key, expected_value|
       subject(:perform) { instance.value_by_resource_key(resource_key) }
 
       it "correctly maps the resource key, #{resource_key}, to the value of the '#{saml_key}' SAML key" do
-        saml_attributes[saml_key] = saml_attributes.delete(resource_key)
+        saml_attributes[saml_key] = saml_attributes.to_h.delete(resource_key)
```

Comment:
After `#to_h`, this delete won't have mutated the attributes. Since there isn't a great way to delete an attribute after constructing the `Attributes` object, what if instead we delayed it until after this mutation?

```suggestion
        saml_attributes[saml_key] = saml_attributes.delete(resource_key)
```

```ruby
let(:instance) { described_class.new(OneLogin::RubySaml::Attributes.new(saml_attributes), attribute_map) }
# ...
let(:saml_attributes) do
  {
    # ...
  }
end
```

Reply:
Other question, does it need to .delete and reassign at all?

When I look at it, I'm not really sure why we're it was originally doing basically:
```
x = {'foo' => 'bar' }
x.delete 'foo'
x['foo]' = 'bar'
```

Okay the OneLogin::RubySaml::Attributes acts kind of like a hash, just without all of the methods and interface; but what was the original test proving with the modification?


(When I did this change, it was basically 'see test failure, do thing fix test failure'; not really understanding why)

Reply:
It's hard to say for sure. I feel like it's testing that the original name *isn't* being used, instead the configured name is being used. Another way of accomplishing this would be to create a new attributes object each time, instead of mutating the shared object.

Reply:
Ah, oops, missed that was was saml_key vs resource_key